### ISSUE TITLE
Feature Request: Hot Reload CSS changes

### ISSUE THREAD

**Issue by acrodrig** [quoid/userscripts#158]
Is is possible to hot reload CSS changes without reloading the entire page to seem have an effect? Thanks for letting me know.

**Comment by quoid** [comment]
@acrodrig 

Thanks for the response and nice words.

>  to iteratively add some CSS to the page

If I am understanding correctly, you are trying to build custom css for the web app and because the load time is relatively long, it's a bit a pain to make changes, view those changes, refresh, etc... 

I can see why hot reloading would be beneficial while creating the userscript, but once that's complete wouldn't the feature become fairly useless to you?  Please correct me if my understanding of how this feature would be a value to you is incorrect.

I am not sure I have the bandwidth right now, with all the others things that need doing, to implement such a feature. To be completely transparent I am a bit ambivalent on the feature as I currently understand it. Fairly soon I hope to move injection to `tabs.insertCSS` and when that is implemented, I am not sure it will even be possible to hot reload css.

That being said I will keep this in mind when I do a new feature push

**Comment by acrodrig** [comment]
@quoid some more context below:

> while creating the userscript, but once that's complete wouldn't the feature become fairly useless to you?

Yes, that is absolutely right.

I see your point about `tabs.insertCSS`. Maybe just a button to force reload? I think even with `tabs.insertCSS` it may be possible to add a timestamp to the URL query string to force reload (something like `https://host/style.css?ts=1234`).

I believe that is what extensions such as [CSS Reloader](https://chrome.google.com/webstore/detail/css-reloader/dnfpcpfijpdhabaoieccoclghgplmpbd?hl=en) do behind the scenes.

To sum it up, the utility is to have in the development phase, when you are understanding the DOM of the target page and adding the CSS/JS to the page is a trial and error process.

If creating extensions for Safari had not become so complicated I would offer a PR.

**Comment by quoid** [comment]
@acrodrig 

> To sum it up, the utility is to have in the development phase, when you are understanding the DOM of the target page and adding the CSS/JS to the page is a trial and error process.

The more I think about this, the more I like this idea. I think I could find value personally. Like I said above, time is short these days, but I promise I will get to this, but it might take longer than your YNAB project.

> If creating extensions for Safari had not become so complicated I would offer a PR.

It truly is a pain. Ever since Safari moved away from the Safari JS api it's been a challenge. The move to Safari Web Extensions is an improvement, but still much more challenging than Firefox or Chrome. I really enjoy certain aspects of the Safari browser which keeps me using it for personal use, or I would have jumped into Firefox full time by now 😄
